ops/pebble.py

```diff
@@ -3211,6 +3317,44 @@ def get_notices(
         resp = self._request('GET', '/v1/notices', query)
         return [Notice.from_dict(info) for info in resp['result']]
 
+    def get_identities(self) -> Dict[str, Identity]:
+        """Get all identities in Pebble.
+
+        Returns:
+            A dict mapping identity names to :class:`Identity` objects.
+        """
+        resp = self._request('GET', '/v1/identities')
+        result = resp['result']
+        return {name: Identity.from_dict(d) for name, d in result.items()}
+
+    def replace_identities(
+        self, identities: Mapping[str, Union[IdentityDict, Identity, None]]
```

Comment:
Which do we propose the charmers to use: `IdentityDict` or `Identity`?

If it's the earlier, I'd be against exposing `'access': Literal['untrusted', 'metrics', 'read', 'admin']` to our charmers.

Recall that a literal is not a constant, it's a verbatim string.

For example, the following will be rejected by the type checker and there's no decent work-around:

```py
id1 = json.loads(json.dumps({"access": "read", ...}))
```

If some charm gets identities over a relation, they won't be able to apply them to pebble in a way that static type checker recognises.

Only hard-coded identities will be OK, and even that with some caveats.

And if the recommended way is `Identity` dataclass, then that should go first.

Reply:
From stand-up: we can always reorder options in a Union. closing this thread for now.